ngmix/tests/test_leastsqbound.py

```diff
@@ -0,0 +1,274 @@
+def test_leastsqbound_smoke(use_prior):
+    rng = np.random.RandomState(2830)
+    ntrial = 10
+    scale = 0.263
+
+    fit_model = 'exp'
+
+    noise = 0.1
+    psf_ngauss = 3
+
+    psf_guesser = ngmix.guessers.CoellipPSFGuesser(
+        rng=rng,
+        ngauss=psf_ngauss,
+    )
+
+    psf_fitter = ngmix.fitting.LMCoellip(ngauss=psf_ngauss)
+
+    psf_runner = ngmix.runners.PSFRunner(
+        fitter=psf_fitter,
+        guesser=psf_guesser,
+        ntry=2,
+    )
+
+    if use_prior:
+        prior = get_prior(
+            fit_model=fit_model,
+            rng=rng,
+            scale=scale,
+            T_range=[-1.0, 1.e3],
+            F_range=[0.01, 1000.0],
+        )
+    else:
+        prior = None
+
+    fitter = ngmix.fitting.LM(model=fit_model, prior=prior)
+    guesser = ngmix.guessers.TFluxGuesser(
+        rng=rng,
+        T=0.25,
+        flux=100.0,
+        prior=prior,
+    )
+
+    runner = ngmix.runners.Runner(
+        fitter=fitter,
+        guesser=guesser,
+        ntry=2,
+    )
+
+    allflags = np.zeros(ntrial)
+    for i in range(ntrial):
+        data = get_model_obs(
+            rng=rng,
+            model='exp',
+            noise=noise,
+        )
+        obs = data['obs']
+
+        try:
+            res = bootstrap(obs=obs, runner=runner, psf_runner=psf_runner)
+            allflags[i] = res['flags']
+        except ngmix.BootPSFFailure:
+            allflags[i] = 1
+
+    assert np.any(allflags == 0)
+
+
+@pytest.mark.parametrize('fracdev_bounds', [None, (0, 1)])
+def test_leastsqbound_bounds(fracdev_bounds):
+    rng = np.random.RandomState(2830)
+
+    ntrial = 10
+    fit_model = 'bd'
+    scale = 0.263
+
+    noise = 0.1
+    psf_ngauss = 3
+
+    psf_guesser = ngmix.guessers.CoellipPSFGuesser(
+        rng=rng,
+        ngauss=psf_ngauss,
+    )
+
+    psf_fitter = ngmix.fitting.LMCoellip(ngauss=psf_ngauss)
+
+    psf_runner = ngmix.runners.PSFRunner(
+        fitter=psf_fitter,
+        guesser=psf_guesser,
+        ntry=2,
+    )
+
+    prior = get_prior(
+        fit_model=fit_model,
+        rng=rng,
+        scale=scale,
+        T_range=[-1.0, 1.e3],
+        F_range=[0.01, 1000.0],
+        fracdev_bounds=fracdev_bounds,
+    )
+
+    guesser = ngmix.guessers.BDFGuesser(
+        T=0.25,
+        flux=100.0,
+        prior=prior,
+    )
+
+    fitter = ngmix.fitting.LM(model=fit_model, prior=prior)
+    runner = ngmix.runners.Runner(
+        fitter=fitter,
+        guesser=guesser,
+        ntry=2,
+    )
+
+    allflags = np.zeros(ntrial)
+    for i in range(ntrial):
+        data = get_model_obs(
+            rng=rng,
+            model='exp',
+            noise=noise,
+        )
+        obs = data['obs']
+
+        try:
+            res = bootstrap(obs=obs, runner=runner, psf_runner=psf_runner)
+            allflags[i] = res['flags']
+        except ngmix.BootPSFFailure:
+            allflags[i] = 1
+
+    assert np.any(allflags != 0)
+
+
+def test_leastsqbound_errors():
```

Comment:
Shouldn't the output flags be tested here or something?

Reply:
yes, just out of time for today

Reply:
ahh sorry to jump the gun! Have a good weekend!